### ISSUE TITLE
Expose a "make awaitable" helper

### ISSUE THREAD

**Issue by maxfischer2781** [maxfischer2781/asyncstdlib#42]
Following some chat and SO questions, it seems worthwhile to offer a "sync to async" helper for ``await ...()``. ``asyncstdlib._core`` already contains one, which can serve as a template.

* Takes any callable and returns an asynchronous callable. For example, ``aprint = helper(print)`` would allow ``await aprint("Hello World")``.
    * MUST be "async neutral", i.e. receiving an async callable will just create another async callable – ``await afoo()`` and ``await helper(afoo)()`` MUST behave the same.

**Comment by ArtyomKozyrev8** [comment]
Hello, I recently found the project and consider it interesting for me. I would like to contribute to the project. I would like to try to implement the feature which described in this issue.

I have some question: 1. what file should I add the feature, should it be `_core.py`? 2. I see that there is `Awaitify` class and separate `force_async`, which already have the desired behavior (sorry if I am wrong), what new functionality you would like to add to them? 

Look forward to get answer from you, if there is some chat about the project, please let me know.

**Comment by maxfischer2781** [comment]
Thanks for wanting to contribute. 👍 

This feature would go into ``asyncstdlib.asynctools``. This is purposely separate from ``asyncstdlib._core`` to allow a richer interface and avoid changes in the public/private helpers breaking each other.

- The most important change is that the new helper should properly provide the metadata of the wrapped function, e.g. ``__doc__``. You can probably use [``functools.wraps``](https://docs.python.org/3/library/functools.html#functools.wraps) for this.  
- Writing a proper docstring would be useful. It doesn't have to be long, just a terse description and a short example is enough.  
- If you are taking ``Awaitify`` as a template/reference, feel free to simplify the new helper from using a class to a function. You can also drop the peek cache (the ``_async_call`` attribute of ``Awaitify``) and check for an awaitable on each call. I've gotten some feedback that the ``Awaitify`` implementation is a tad hard to follow (\*cough\*), so for a public facing helper that is likely to get inspected by people wanting to use it a more straightforward implementation is fine.

I have just created a [gitter channel](https://gitter.im/maxfischer2781/asyncstdlib) for this project. Feel free to drop by to discuss this and get some pointers.

**Comment by ArtyomKozyrev8** [comment]
@maxfischer2781 ok I got it, I forked the project and will try to implement the feature.

**Comment by maxfischer2781** [comment]
Closed by #46.
